docs/contributing/advanced_development_topics.rst

```diff
@@ -0,0 +1,156 @@
+
+- Add ``random_state`` argument to visualizer.
+
+    If the visualizer uses/wraps a utility that also has ``random_state``, then the visualizer itself needs to also have this argument which defaults to ``None`` and is passed to all internal stochastic behaviors. This ensures that image comparison testing will work and that users can get repeated behavior from visualizers.
+
+- Use ``np.unique`` instead of `set`.
+
+    If you need the unique values from a list or array, we prefer to use numpy methods wherever possible. We performed some limited benchmarking and believe that ``np.unique`` is a bit faster and more efficient.
+
+- Use sklearn underscore suffix for learned parameters.
+
+    Any parameters that are learned during ``fit()`` should only be added to the visualizer when ``fit()`` is called (this is also how we determine if a visualizer is fitted or not) and should be identified with an underscore suffix. For example, in classification visualizers, the classes can be either passed in by the user or determined when they are passed in via fit, therefore it should be ``self.classes_``. This is also true for other learned parameters, e.g. ``self.score_``, even though this is not created during ``fit()``.
+
+Testing Conventions
+~~~~~~~~~~~~~~~~~~~
+
+- Ensure there is an image comparison test.
+
+    Ensure there is at least one image comparison test per visualizer. This is the primary regression testing of Yellowbrick and these tests catch a lot when changes occur in our dependencies or environment.
+
+- Use pytest assertions rather than unittest.
+
+    We prefer ``assert 2+2 == 4`` rather than ``self.assertEquals(2+2, 4)``. Though there is a lot of legacy unittest assertions, we've moved to pytest and one day believe we will have removed the unittest dependency.
+
+- Use test fixtures and sklearn dataset generators.
+
+    Data is the key to testing with Yellowbrick; often the test package will have fixtures in ``conftest.py`` that can be directly used (e.g. binary vs. multiclass in the ``test_classifier`` package). If one isn't available feel free to use randomly generated datasets from the ``sklearn.datasets`` module e.g. ``make_classification``, ``make_regression``, or ``make_blobs``. For integration testing, please feel free to use one of the Yellowbrick datasets.
+
+- Fix all ``random_state`` arguments.
+
+    Be on the lookout for any method (particularly sklearn methods) that have a ``random_state`` argument and be sure to fix them so that tests always pass!
+
+- Test a variety of inputs.
+
+    Machine learning can be done on a variety of inputs for ``X`` and ``y``, ensure there is a test with numpy arrays, pandas DataFrame and Series objects, and with Python lists.
+
+- Test that ``fit()`` returns self.
```

Comment:
I feel like we should implement a test helper function for this.

Reply:
We have the checks module framework from #180 that was supposed to do this for us, but we didn't get too far on it.